src/System.Memory/src/System/Buffers/SequenceReader.cs

```diff
@@ -75,36 +74,25 @@ public ReadOnlySpan<T> UnreadSpan
         /// <summary>
         /// The total number of <typeparamref name="T"/>'s processed by the reader.
         /// </summary>
-        public long Consumed { get; private set; }
+        public long Consumed { readonly get; private set; }
 
         /// <summary>
         /// Remaining <typeparamref name="T"/>'s in the reader's <see cref="Sequence"/>.
         /// </summary>
-        public long Remaining => Length - Consumed;
+        public readonly long Remaining => Length - Consumed;
 
         /// <summary>
         /// Count of <typeparamref name="T"/> in the reader's <see cref="Sequence"/>.
         /// </summary>
-        public long Length
-        {
-            get
-            {
-                if (_length < 0)
-                {
-                    // Cache the length
-                    _length = Sequence.Length;
-                }
-                return _length;
-            }
-        }
+        public long Length { readonly get; private set; }
```

Comment:
Can this instead be:
```C#
public readonly long Length { get; }
```
?  It looks like it's only set in the ctor, in which case this should be fine.

Reply:
I'll give that a shot.